src/main/java/roomescape/controller/ReservationController.java

```diff
@@ -0,0 +1,55 @@
+package roomescape.controller;
+
+import jakarta.validation.Valid;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import roomescape.dto.RequestCreateReservation;
+import roomescape.domain.Reservation;
+import roomescape.dto.ResponseReservation;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+
+@RestController
+public class ReservationController {
+
+    private final List<Reservation> reservations = new ArrayList<>();
+    private final AtomicLong index = new AtomicLong(1);
+
+    @PostMapping("/reservations")
+    public ResponseEntity<ResponseReservation> createReservation(
+        @Valid @RequestBody RequestCreateReservation requestCreateReservation
+    ) {
+        Reservation reservation = requestCreateReservation.toReservation(index.getAndIncrement());
+        reservations.add(reservation);
+
+        return ResponseEntity
+            .created(URI.create("/reservations/" + reservation.getId()))
+            .body(ResponseReservation.of(reservation));
+    }
+
+    @GetMapping("/reservations")
+    public ResponseEntity<List<ResponseReservation>> getReservations() {
+        List<ResponseReservation> reservationList = reservations.stream().map(ResponseReservation::of).toList();
```

Comment:
```suggestion
        List<ResponseReservation> reservationList = reservations.stream()
            .map(ResponseReservation::of)
            .toList();
```
개행을 통해서 가독성을 높여주면 좋을 것 같아요